### PR TITLE
Fix Reference in TabletGroupWatcher

### LINKED ISSUE #2763: Invalid Path in TabletGroupWatcher
The deleteRows test in ShellServerIT is failing due to recent change #2750 
<pre>
2022-06-09T10:42:04,829 [manager.EventCoordinator] INFO : Merge state of 1k;row7;row5 set to MERGING
2022-06-09T10:42:04,829 [manager.Manager] DEBUG: Deleting tablets for 1k;row7;row5
2022-06-09T10:42:04,830 [manager.Manager] DEBUG: Found following tablet 1k<;row7
2022-06-09T10:42:04,830 [manager.Manager] DEBUG: Making file deletion entries for 1k;row7;row5
2022-06-09T10:42:04,831 [manager.Manager] DEBUG: Finished gathering information from 1 of 1 servers in 0.00 seconds
2022-06-09T10:42:04,833 [manager.Manager] ERROR: Unable merge metadata table records
org.apache.accumulo.core.client.AccumuloException: java.lang.IllegalArgumentException: Invalid path provided, no scheme in t-000008q
        at org.apache.accumulo.manager.TabletGroupWatcher.deleteTablets(TabletGroupWatcher.java:694) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.TabletGroupWatcher.updateMergeState(TabletGroupWatcher.java:594) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.TabletGroupWatcher.run(TabletGroupWatcher.java:341) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
Caused by: java.lang.IllegalArgumentException: Invalid path provided, no scheme in t-000008q
        at org.apache.accumulo.core.metadata.ValidationUtil.validate(ValidationUtil.java:47) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.core.metadata.ValidationUtil.validate(ValidationUtil.java:41) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.server.metadata.ServerAmpleImpl.createDeleteMutation(ServerAmpleImpl.java:208) ~[accumulo-server-base-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.server.metadata.ServerAmpleImpl.putGcFileAndDirCandidates(ServerAmpleImpl.java:136) ~[accumulo-server-base-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.TabletGroupWatcher.deleteTablets(TabletGroupWatcher.java:666) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        ... 2 more
</pre>

### PR DESCRIPTION
**Pull request by milleruntime**
* Closes #2763
* Also fix the misplaced private String member tabletDir

### CODE REVIEW

**Comment by milleruntime** [comment]
The ITs that were failing are all passing with this fix (CloneTestIT,DeleteRowsSplitIT,DeleteRowsIT,ShellServerIT)

**Review by EdColeman — APPROVED** [review]

**Comment by EdColeman** [comment]
I merged this into a local branch that I'm working on.  With these changes, sunny tests pass.  I kicked off a set of full ITs and it not looking as good.  4 of the tasks failed, and 4 remaining are just spinning.  Of the the ones that failed (so far) 

Three of them failed with errors similar to:

```
MultiTableInputFormatIT>AccumuloClusterHarness.setupCluster:167 » IllegalState Unable to find instance id from zookeeper
```

And one timed out:

```
ShellServerIT.deleterows execution timed out after 60000 ms
```

**Review by dlmarion — APPROVED** [review]

**Comment by milleruntime** [comment]
> ShellServerIT.deleterows execution timed out after 60000 ms

This was the test that was failing prior to this fix. Do you still have the logs?

File: core/src/main/java/org/apache/accumulo/core/gc/Reference.java [hunk 1]
```diff
@@ -36,7 +35,6 @@ public Reference(TableId tableId, String metadataEntry) {
     MetadataSchema.TabletsSection.ServerColumnFamily.validateDirCol(tableId.canonical());
     this.tableId = tableId;
     this.metadataEntry = metadataEntry;
-    this.tabletDir = metadataEntry;
   }
```
[the remark this answers is not in the document]
Comment (milleruntime) [reply core/src/main/java/org/apache/accumulo/core/gc/Reference.java:38]: Yeah, this is where we collect the references in the GC:
https://github.com/apache/accumulo/blob/main/server/gc/src/main/java/org/apache/accumulo/gc/GCRun.java#L149-L156

This chunk of code is confusing to me and I am currently working on PR to break it up into multiple lines with comments.


> In fact, you may be able to remove ReferenceDirectory if you have the ability to determine if the metadata entry is a directory.

I had originally created this class as a concrete way to differentiate between the two types here:
https://github.com/apache/accumulo/blob/37e1c6f275cca0b118ed6ab631345c0699515552/server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectionAlgorithm.java#L142

But I could make a `isDirectory()` method that gets overridden.

**Comment by milleruntime** [comment]
Since you guys both approved this fix, I am going to merge. But I have a follow up with more GC clean up on the way...

File: core/src/main/java/org/apache/accumulo/core/gc/Reference.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (milleruntime) [reply core/src/main/java/org/apache/accumulo/core/gc/Reference.java:38]: Follow up changes in https://github.com/apache/accumulo/pull/2767

**Pull request merged**